mathesar/api/db/viewsets/queries.py

```diff
@@ -30,15 +31,37 @@ class QueryViewSet(
     pagination_class = DefaultLimitOffsetPagination
     filter_backends = (filters.DjangoFilterBackend,)
     filterset_class = UIQueryFilter
+    permission_classes = [IsAuthenticatedOrReadOnly]
     access_policy = QueryAccessPolicy
 
     def get_queryset(self):
-        queryset = self.access_policy.scope_queryset(self.request, UIQuery.objects.all())
+        queryset = None
+        queryset = self._get_scoped_queryset()
         schema_id = self.request.query_params.get('schema')
         if schema_id:
             queryset = queryset.filter(base_table__schema=schema_id)
         return queryset.order_by('-created_at')
 
+    def _get_scoped_queryset(self):
+        """
+        Returns a properly scoped queryset.
+
+        Access to queries may require different access controls, some of which
+        require scoping while others do not. See
+        `QueryAccessPolicy.get_should_queryset_be_scoped` docstring for more
+        information.
+        """
+        should_queryset_be_scoped = \
+            QueryAccessPolicy.get_should_queryset_be_scoped(self.action)
+        if should_queryset_be_scoped:
+            queryset = self.access_policy.scope_queryset(
+                self.request,
+                UIQuery.objects.all()
+            )
+        else:
+            queryset = UIQuery.objects.all().order_by('-created_at')
```

Comment:
I noticed that the `order_by` transformations is applied here (which is not useful, I think), but not on the scoped queryset. Is this an oversight?

Reply:
@dmos62 

> but not on the scoped queryset. Is this an oversight?

The `get_queryset` method applies a `order_by` transformation when returning the queryset. So, this is handled for both cases.
```python
    def get_queryset(self):
        queryset = self._get_scoped_queryset()
        schema_id = self.request.query_params.get('schema')
        if schema_id:
            queryset = queryset.filter(base_table__schema=schema_id)
        return queryset.order_by('-created_at')
```

I added a commit to remove the redundant `order_by` transformation above: https://github.com/centerofci/mathesar/pull/3113/commits/bb41400ec9609ba0108dd51da243883d7b896737